src/lib/internal_tracing/context_call/internal_tracing_context_call.ml

```diff
@@ -2,12 +2,14 @@ open Core_kernel
 
 let last_call_id = ref 0
 
-let key = Univ_map.Key.create ~name:"call_id" sexp_of_int
+let key = Univ_map.Key.create ~name:"call_id" [%sexp_of: int * string]
 
-let with_call_id f =
+let with_call_id ?(tag = "") f =
```

Comment:
Would it be better to use `?tag` without a default, so that it's `None` if omitted?

Reply:
That works too, main reasoning behind just having an empty string there is so that the property is always there, and that the logic for having it on the output or not doesn't show up in two places (because of the current duplication in how these special control checkpoints are handled).